### PR TITLE
Created sink parallel execution (#624)

### PR DESCRIPTION
**Pull request by MattSzm**
This is a proof of concept. Please validate the idea before I write all methods implementations/tests because for me the issue description can be interpreted in a couple of ways.

### CODE REVIEW

**Comment by paualarco** [comment]
> This is a proof of concept. Please validate the idea before I write all methods implementations/tests because for me the issue description can be interpreted in a couple of ways.

That was the idea @MattSzm, I like it :)

**Comment by paualarco** [comment]
Will do a release after once merged, with all these recent mongo improvements

**Review by paualarco — APPROVED** [review]
Nice job 🥇

**Comment by MattSzm** [comment]
Hi @paualarco I found `should update many documents per each received request whereas requests are grouped in lists. Requests from a single group are executed at once in parallel.` test being flaky.
```
[info]   60 was not equal to 40 (MongoSinkSuite.scala:294)
```

I've added the commit which fixes this. Fortunately, it was a problem with the test - not with the feature.
What happened:
```
    val e1 = genEmployeesWith(name = Some(name1), n = 10).sample.get
    val e2 = genEmployeesWith(name = Some(name2), age = Some(31), n = 20).sample.get
    val e3 = genEmployeesWith(name = Some(name3), n = 30).sample.get

    val u1 = (Filters.eq("name", name1), Updates.set("name", name3))
    val u2 = (Filters.eq("name", name2), Updates.combine(Updates.set("name", name1), Updates.inc("age", 10)))
```

In the test case, we run these updates in parallel, so the results depend on the order.
1. u1 first and u2 second -> we have 40 items with name3 (10 + 30)
2. u2 first and u1 second -> we change items with name2 to name1 and next we change all items with name1 to name3 which gave us 60 items with name3.

I'm sorry for the mistake. But, we can be sure now, that these calls are made in parallel haha :D

File: mongodb/src/it/scala/monix/connect/mongodb/MongoSinkSuite.scala [hunk 1]
```diff
@@ -273,15 +273,16 @@ class MongoSinkSuite extends AsyncFlatSpec with MonixTaskSpec with Fixture with
 
   "updateManyPar" should "update many documents per each received request whereas requests are grouped in lists. " +
     "Requests from a single group are executed at once in parallel." in {
-    val name1 = "Name1"
-    val name2 = "Name2"
-    val name3 = "Name3"
+    val name1 = "Name4"
+    val name2 = "Name5"
+    val name3 = "Name6"
+    val name4 = "Name7"
```
Comment (paualarco) [inline mongodb/src/it/scala/monix/connect/mongodb/MongoSinkSuite.scala:279]: Why does the string value does not match with the variable name?
```scala
    val name1 = "Name1"
    val name2 = "Name2"
    val name3 = "Name3"
    val name4 = "Name4"
```

**Comment by paualarco** [comment]
@MattSzm yep there is some flaky test that need fixing... the failures are mainly related to random generators in combination with parallel tests. I will merge your pr after addressing my comment :)

File: mongodb/src/it/scala/monix/connect/mongodb/MongoSinkSuite.scala (hunk 1 above)
Comment (MattSzm) [reply mongodb/src/it/scala/monix/connect/mongodb/MongoSinkSuite.scala:279]: Actually, it does not have any purpose. Personally, I'm not a fan of using the same names multiple times in multiple test cases (`updateMany` has names from 1 to 3). But, I'm making a commit with your suggestion applied.

Comment (paualarco) [reply mongodb/src/it/scala/monix/connect/mongodb/MongoSinkSuite.scala]: Me neither, that's why we should start using generators in combination with prefixes in all our test suites. 
I don't mind if it's 1 2 3 or 4 3 5, but the var name and value should be in-sync

**Review by paualarco — APPROVED** [review]

**Pull request merged**